### PR TITLE
Add booster chat app example

### PR DESCRIPTION
**Pull request by juanjoman**
This is a simple chat app made with Booster with 3 models:  Chat Room, Users and Messages

**Note:** yarn.lock includes a lot of lines. Ignore it

### CODE REVIEW

File: booster-chat-app/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Simply chat server powered by Booster Framework
```
Comment (juanjoman) [inline booster-chat-app/README.md:4]: I will create a quick PR later to add the article link

File: booster-chat-app/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+This project is a sample implementation of a chat server in [Booster](https://booster.cloud) to serve as a guidance of the main concepts and artifacts used in the framework.
+
+## Chat requirements
+
+We are going to structure our chat app in chat rooms, which are going to be created or modified only by admins. Inside each chat room the user sends messages to chat with other users.
```
Comment (itrion) [inline booster-chat-app/README.md]: By looking at the code, I understand that chat rooms can be created by both, users and admins, and a chat room can be modified only by its owner

File: booster-chat-app/src/event-handlers/HandleUserJoined.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { MessageType } from '../entities/Message'
+import { ChatRoom } from '../entities/ChatRoom'
+
+@EventHandler(UserJoined)
+export class HandleUserJoined {
+  public static async handle(event: UserJoined, register: Register): Promise<void> {
+    const chatRoom = await Booster.fetchEntitySnapshot(ChatRoom, event.id)
+    if (chatRoom) {
+      register.events(
+        new MessageCreated(event.id, chatRoom.id, event.user, `${event.user.alias} has joined in`, MessageType.JOIN)
```
Comment (itrion) [inline booster-chat-app/src/event-handlers/HandleUserJoined.ts:14]: 😎  👍

File: booster-chat-app/src/entities/Message.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { UUID } from '@boostercloud/framework-types'
+import { UserProfile } from './UserProfile'
+
+export enum MessageType {
+  CHAT = 'chat',
+  JOIN = 'join',
+}
+
+export class Message {
```
Comment (itrion) [inline booster-chat-app/src/entities/Message.ts:11]: Is this working without the `@Entity` annotation?

File: booster-chat-app/src/read-models/MessageReadModel.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    public id: UUID,
+    readonly chatRoomId: UUID,
+    readonly body: string,
+    readonly messageType: MessageType,
+    readonly createdAt: string,
+    readonly createdBy: UserProfile
+  ) {}
+
+  @Projects(Message, 'chatRoomId')
+  public static onMessageCreated(message: Message): MessageReadModel {
```
Comment (itrion) [inline booster-chat-app/src/read-models/MessageReadModel.ts:22]: This is related to my previous comment about the Message class missing the Entity decorator. Is this code actually being called?

File: booster-chat-app/src/entities/Message.ts (hunk 4 above)
Comment (juanjoman) [reply booster-chat-app/src/entities/Message.ts:11]: Whoops I missed that! Thank you!!

File: booster-chat-app/README.md (hunk 2 above)
Comment (juanjoman) [reply booster-chat-app/README.md]: You're right, they are created by anyone but modified by the admin. Nice catch 👏

**Pull request merged**